product_tax_multicompany_default/models/product.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2017 Carlos Dauden - Tecnativa <carlos.dauden@tecnativa.com>
+# Copyright 2018 Vicent Cubells - Tecnativa <vicent.cubells@tecnativa.com>
+# Copyright 2023 Eduardo de Miguel - Moduon <edu@moduon.team>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from typing import List
+
+from odoo import api, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    def taxes_by_company(self, field, company, match_tax_ids=None):
+        taxes_ids = []
+        if match_tax_ids is None:
+            taxes_ids = company[field].ids
+        # If None: return default taxes
+        if not match_tax_ids:
+            return taxes_ids
+        AccountTax = self.env["account.tax"]
+        for tax in AccountTax.browse(match_tax_ids):
+            taxes_ids.extend(
+                AccountTax.search(
+                    [("name", "=", tax.name), ("company_id", "=", company.id)]
+                ).ids
+            )
+        return taxes_ids
+
+    def _delete_product_taxes(
+        self,
+        excl_customer_tax_ids: List[int] = None,
+        excl_supplier_tax_ids: List[int] = None,
+    ):
+        """Delete taxes from product excluding chosen taxes
+
+        :param excl_customer_tax_ids: Excluded customer tax ids
+        :param excl_supplier_tax_ids: Excluded supplier tax ids
+        """
+        tax_where = " AND tax_id NOT IN %s"
+        # Delete customer taxes
+        customer_sql = "DELETE FROM product_taxes_rel WHERE prod_id IN %s"
```

Comment:
Why use SQL?
Set customer_tax_ids as you like before than this code:
```
        self.write(
            {
                "taxes_id": [(6, 0, customer_tax_ids)],
                "supplier_taxes_id": [(6, 0, supplier_tax_ids)],
            }
        )

```

Reply:
Cannot do it with sudo or SUPERUSER_ID, don't know why, but I've tryied a lot of possibilities with sudo, with_user(SUPERUSER_ID), with_context(allowed_company_ids=...), with_company(), and nothing...

Systematically is excluing all the [(5, None)] on other companies. 
This is the reason why I've chosen to empty vía sql